src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Tables/TaskInfo.sql

```diff
@@ -19,6 +19,7 @@ CREATE TABLE [dbo].[TaskInfo] (
     [EndDateTime]       DATETIME2 (7) NULL,
     [Worker]            VARCHAR (100) NULL,
     [RestartInfo]       VARCHAR (MAX) NULL,
+    [ParentTaskId]      VARCHAR (64)  NULL,
     CONSTRAINT PKC_TaskInfo PRIMARY KEY CLUSTERED (TaskId) WITH (DATA_COMPRESSION = PAGE)
 ) ON [PRIMARY] TEXTIMAGE_ON [PRIMARY];
 
```

Comment:
There are several problems with this table design which prevent using it as a generic queue solution long term:
1. Ids are character fields filled by GUIDs.  This will cause problems with large table sizes.
2. There is no partitioning so this will limit queue throughput to ~100 operations per second.
3. QueueId that can be used as queue type (import. export, etc) is not an enumeration, and table is not physically partitioned on it, so there are performance affects between queues.
4. TaskContext, designed to store internal task state, leads to N-square problem, where N - number of internal states. We should encourage relational design and split large task into N simple tasks and use their Result for internal states, this way expenses are always ~N.

Reply:
Agree, as offline sync, we would work on the task improvement separately.